server/src/main/java/org/opensearch/cluster/service/ClusterManagerTaskThrottler.java

```diff
@@ -108,10 +108,12 @@ public boolean isThrottlingEnabled() {
     }
 
     void validateSetting(final Settings settings) {
-        if (minNodeVersionSupplier.get().compareTo(Version.V_2_4_0) < 0) {
-            throw new IllegalArgumentException("All the nodes in cluster should be on version later than or equal to 2.4.0");
-        }
         Map<String, Settings> groups = settings.getAsGroups();
+        if (groups.size() > 0) {
```

Comment:
For my understanding what does this validate?

Reply:
> When other settings get changes this validate method gets called with empty settings. So that's why added check to avoid version check when called with empty setting.

@dhwanilpatel can you point to code where this happens?

Reply:
Will need to go through the code to find out the pointers, I have seen this behavior in local testing.

Reply:
Here is the code reference where it validates setting against all settings before applying it. 

Code ref: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/settings/AbstractScopedSettings.java#L167-L188